src/tests/Microsoft.PowerFx.Core.Tests/SymbolTableTests.cs

```diff
@@ -131,5 +132,42 @@ public void ValidateNames()
 
             Assert.Throws<ArgumentException>(() => s1.AddConstant(invalidName, FormulaValue.New(3)));
         }
+
+        [Fact]
+        public void MutableSupportedFunctionsTest()
+        {
+            var symbolTableOriginal = new Engine(new PowerFxConfig()).SupportedFunctions;
+            var symbolTableCopy1 = symbolTableOriginal.GetMutableSupportedFunctionsCopy();
+            var symbolTableCopy2 = symbolTableOriginal.GetMutableSupportedFunctionsCopy();
+
+            var originalCount = symbolTableOriginal.Functions.Count();
+            var copyCount1 = symbolTableCopy1.Functions.Count();
+            var copyCount2 = symbolTableCopy2.Functions.Count();
+
+            Assert.Equal(copyCount1, originalCount);
+            Assert.Equal(copyCount2, originalCount);
+
+            symbolTableCopy1.RemoveFunction("Abs");
+            symbolTableCopy2.RemoveFunction("Day");
+
+            Assert.Equal(originalCount, symbolTableOriginal.Functions.Count());
+
+            Assert.NotEqual(copyCount1, symbolTableCopy1.Functions.Count());
+            Assert.NotEqual(copyCount2, symbolTableCopy2.Functions.Count());
+
+            Assert.Contains(symbolTableOriginal.Functions, f => f.Name == "Abs");
+            Assert.Contains(symbolTableOriginal.Functions, f => f.Name == "Day");
+            Assert.Contains(symbolTableOriginal.Functions, f => f.Name == "Text");
+            Assert.Contains(symbolTableOriginal.Functions, f => f.Name == "Value");
+            Assert.Contains(symbolTableCopy1.Functions, f => f.Name == "Day");
+            Assert.Contains(symbolTableCopy1.Functions, f => f.Name == "Text");
+            Assert.Contains(symbolTableCopy1.Functions, f => f.Name == "Value");
+            Assert.Contains(symbolTableCopy2.Functions, f => f.Name == "Abs");
+            Assert.Contains(symbolTableCopy2.Functions, f => f.Name == "Text");
+            Assert.Contains(symbolTableCopy2.Functions, f => f.Name == "Value");
+
+            Assert.DoesNotContain(symbolTableCopy1.Functions, f => f.Name == "Abs");
+            Assert.DoesNotContain(symbolTableCopy2.Functions, f => f.Name == "Day");
+        }
```

Comment:
another test:
1. Things that shouldn't be copied, aren't. Ie, if we only copy functions, then the vars aren't copied.

Reply:
There are no public properties (apart from functions) that can be compared.

Reply:
There is a public SymbolNames property. 

And the test can access internal surface area.

Reply:
For some reason it was not being displayed in the intellisense. Adding new test case.